models/issue_indexer.go

```diff
@@ -0,0 +1,201 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+
+package models
+
+import (
+	"os"
+	"strconv"
+
+	"github.com/blevesearch/bleve"
+	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/modules/log"
+	"github.com/blevesearch/bleve/analysis/analyzer/simple"
+	"code.gitea.io/gitea/modules/util"
+	"github.com/blevesearch/bleve/search/query"
+	"strings"
+)
+
+// indexerUpdateQueue queue of issues that need to be updated in the issues
+// indexer
+var indexerUpdateQueue chan *Issue
+
+// IssueIndexerData issue data stored in the issues indexer
+type IssueIndexerData struct {
+	ID int64
+	RepoID int64
+
+	Title string
+	Content string
+}
+
+// numericQuery an numeric-equality query for the given value and field
+func numericQuery(value int64, field string) *query.NumericRangeQuery {
+	f := float64(value)
+	tru := true
+	q := bleve.NewNumericRangeInclusiveQuery(&f, &f, &tru, &tru)
+	q.SetField(field)
+	return q
+}
+
+// SearchIssuesByKeyword searches for issues by given conditions.
+// Returns the matching issue IDs
+func SearchIssuesByKeyword(repoID int64, keyword string) ([]int64, error) {
+	fields := strings.Fields(strings.ToLower(keyword))
+	indexerQuery := bleve.NewConjunctionQuery(
+		numericQuery(repoID, "RepoID"),
+		bleve.NewDisjunctionQuery(
+			bleve.NewPhraseQuery(fields, "Title"),
+			bleve.NewPhraseQuery(fields, "Content"),
+		))
+	search := bleve.NewSearchRequestOptions(indexerQuery, 2147483647, 0, false)
+	search.Fields = []string{"ID"}
+
+	index, err := bleve.Open(setting.IssueIndexerPath)
+	if err != nil {
+		return nil, err
+	}
+	result, err := index.Search(search)
+	if err != nil {
+		index.Close()
+		return nil, err
+	}
+	if err = index.Close(); err != nil {
+		return nil, err
+	}
+
+	issueIDs := make([]int64, len(result.Hits))
+	for i, hit := range result.Hits {
+		issueIDs[i] = int64(hit.Fields["ID"].(float64))
+	}
+	return issueIDs, nil
+}
+
+// InitIssuesIndexer initialize issue indexer
+func InitIssuesIndexer() error {
+	_, err := os.Stat(setting.IssueIndexerPath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			if err = CreateIssuesIndexer(); err != nil {
+				return err
+			}
+			go func() {
+				if err := PopulateIssuesIndexer(); err != nil {
```

Comment:
I prefer remove the `go` keyword and then this should be called when Gitea start. And if there are err, it should stop and print the error msg on console.